mlir/lib/Transforms/RemoveDeadValues.cpp

```diff
@@ -178,6 +181,12 @@ static void cleanSimpleOp(Operation *op, RunLivenessAnalysis &la) {
   if (!isMemoryEffectFree(op) || hasLive(op->getResults(), la))
     return;
 
+  for (OpOperand &use : op->getUses())
+    if (auto callOp = dyn_cast<CallOpInterface>(use.getOwner()))
+      if (auto funcOp = dyn_cast<FunctionOpInterface>(callOp.resolveCallable()))
+        if (funcOp.isPublic())
+          return;
```

Comment:
How does this work transitively though?

Reply:
I see `llvm.mlir.undef` and `spirv.Undef`. Is there a generic "undef" value that I can use here?

Reply:
@joker-eph ping